framework/tlvf/src/include/tlvf/BaseClass.h

```diff
@@ -29,6 +30,14 @@ class BaseClass {
     size_t getLen();
     bool isInitialized();
     virtual void class_swap() = 0;
+    template <class T> std::shared_ptr<T> class_cast()
```

Comment:
better have this addition to the signature:

```suggestion
    template <class T> std::shared_ptr<T> class_cast()
    typename std::enable_if<std::is_base_of<BaseClass, T>::value>::type
```

That way this cast would only be available for derived classes.
Check my syntax as it might be a bit off.